### DISCUSSION TITLE
Beetle v0.3.0: Integration and Testing with Tumblebug and Honeybee

### DISCUSSION THREAD

**Discussion by yunkon-kim** [cloud-barista/cm-beetle#132] [Show and tell]
The following link shares the integration and test of Beetle, Tumblebug, Honeybee and model.
: https://github.com/cloud-barista/cm-beetle/blob/main/docs/beetle-v0.3.0-integration-and-testing-with-tumblebug-honeybee-and-model.md

> [!NOTE]
> Since Discussion does NOT support a link to the **Markdown Header**, I just wrote it as a separate document and added it as a link.  

### Envrionment
* Beetle v0.3.0
* cm-model v0.0.3 (Damselfly v0.3.0)
* Honeybee v0.3.0
* Tumblebug v0.10.3 (Spider v0.10.0, CB-MapUI v0.10.0)

### Scenario
1. Get a source group list via Honeybee
1. Get the refined source computing infra info via Honeybee
    * Refined source computing infra info = on-premise model (a.k.a computing infra source model)
    * Used 2 servers info as discussed (i.e., web, nfs)
1. Recommend a target model for computing infra via Beetle
1. Migrate the computing infra as defined in the target model via Beetle
1. Delete the migrated computing infra via Beetle

👀 Please see the link for more detailed information about the test.

> [!Note]
>  The Honeybee has been providing the always-running servers. The server has been used.

**Comment by yunkon-kim** [comment]
Honeybee에서 지원해 주신 서버를 활용해 Beetle, Honeybee, model 간 테스트를 완료하였고 이를 공유 드립니다 :-) (Thanks @cloud-barista/cm-honeybee-maintainer)

**Reply by ish-hcc** [reply]
@yunkon-kim 감사합니다! 수고 많으셨습니다 :)

**Comment by yunkon-kim** [comment]
@ish-hcc (@sykim-etri) 안녕하세요.

현재, Beetle v0.2.2에서는 목표 컴퓨팅 인프라 추천, 컴퓨팅 인프라 마이그레이션/조회/삭제 API를 제공해 드리고 있습니다.

**Cicada**의 컴퓨팅 인프라 마이그레이션 워크플로우 기능, API 등을 업데이트하시는데
위 내용과 Beetle [v0.2.2 release](https://github.com/cloud-barista/cm-beetle/releases/tag/v0.2.2) 를 참고하시면 될 것 같습니다.

Beetle을 통해 마이그레이션된 인프라를 조회하실 수 있으며,
이 밖에 **Grasshopper**에서 필요한 기능/API는 Tumblebug API를 활용하시면 되겠습니다.

Tumblebug endpoint를 직접 활용하셔도 되고요. Beetle에서 제공하는 Reverse proxy를 통해 활용하실 수 있습니다.
편하신 방법으로 활용하시면 될 것 같습니다.

Reverse proxy 관련 사항은 https://github.com/cloud-barista/cm-beetle/discussions/105
"5. Use the other APIs to manage Multi-Cloud Infrastructure (MCI)" 를 참고하시기 바랍니다.

진행하시면서 궁금하신 사항 또는 이슈가 있으시면 편하게 답글 납겨주시기 바랍니다 :-)
